src/components/calcite-checkbox/calcite-checkbox.tsx

```diff
@@ -45,116 +74,125 @@ export class CalciteCheckbox {
     // prevent duplicate click events that occur
     // when the component is wrapped in a label and checkbox is clicked
     if (
-      (this.el.closest("label") && target === this.inputProxy) ||
-      (!this.el.closest("label") && target === this.el)
+      (this.el.closest("label") && target === this.input) ||
+      (!this.el.closest("label") && currentTarget === this.el)
     ) {
       this.toggle();
     }
   }
 
   @Listen("keydown") keyDownHandler(e: KeyboardEvent) {
     const key = getKey(e.key);
-    if (key === " " || key === "Enter") {
+    if (key === " ") {
       e.preventDefault();
       this.toggle();
     }
   }
 
-  @Watch("checked") checkedWatcher() {
-    this.calciteCheckboxChange.emit();
-    this.checked
-      ? this.inputProxy.setAttribute("checked", "")
-      : this.inputProxy.removeAttribute("checked");
+  @Listen("mouseenter")
+  mouseenter() {
+    this.hovered = true;
+  }
+
+  @Listen("mouseleave")
+  mouseleave() {
+    this.hovered = false;
   }
 
-  private inputProxy: HTMLInputElement;
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
 
   connectedCallback() {
-    this.setupProxyInput();
+    this.renderHiddenCheckboxInput();
     let scale = ["s", "m", "l"];
     if (!scale.includes(this.scale)) this.scale = "m";
   }
 
-  disconnectedCallback() {
-    this.observer.disconnect();
-  }
-
-  componentWillRender() {
-    this.syncProxyInputToThis();
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private renderHiddenCheckboxInput() {
+    this.input = document.createElement("input");
+    this.input.checked = this.checked;
+    this.input.disabled = this.disabled;
+    this.input.onblur = () => (this.focused = false);
+    this.input.onfocus = () => (this.focused = true);
+    this.input.name = this.name;
+    this.input.type = "checkbox";
+    if (this.value) {
+      this.input.value = this.value;
+    }
+    this.el.appendChild(this.input);
   }
 
-  private readonly indeterminatePath = "M4 7h8v2H4z";
-  private readonly checkedPath =
-    "M12.753 3l-7.319 7.497L3.252 8.31 2 9.373l3.434 3.434L14 4.24z";
-
-  private getPath = (): string =>
-    this.indeterminate
-      ? this.indeterminatePath
-      : this.checked
-      ? this.checkedPath
-      : "";
-
   render() {
     return (
-      <Host
-        role="checkbox"
-        aria-checked={this.checked.toString()}
-        tabindex={this.disabled ? "-1" : "0"}
-      >
+      <Host role="checkbox" aria-checked={this.checked.toString()}>
         <svg class="check-svg" viewBox="0 0 16 16">
           <path d={this.getPath()} />
         </svg>
         <slot />
       </Host>
     );
   }
-
-  private setupProxyInput() {
```

Comment:
So does this completely remove the proxy inputs?

Reply:
Correct, yes.